cmd/snap-confine/snap-confine-invocation.c

```diff
@@ -66,3 +71,18 @@ void sc_init_invocation(sc_invocation *inv, const struct sc_args *args, const ch
     debug("confinement:  %s", inv->classic_confinement ? "classic" : "non-classic");
     debug("base snap:    %s", inv->base_snap_name);
 }
+
+void sc_apply_invocation_fallback(sc_invocation *inv) {
+    /* As a special fallback, allow the base snap to degrade from "core" to
+     * "ubuntu-core". This is needed for the migration tests. */
+    char mount_point[PATH_MAX] = {0};
+    sc_must_snprintf(mount_point, sizeof mount_point, "%s/%s/current/", SNAP_MOUNT_DIR, inv->base_snap_name);
+
+    if (sc_streq(inv->base_snap_name, "core") && access(mount_point, F_OK) != 0) {
+        sc_must_snprintf(mount_point, sizeof mount_point, "%s/%s/current/", SNAP_MOUNT_DIR, "ubuntu-core");
+        if (access(mount_point, F_OK) == 0) {
+            inv->base_snap_name = "ubuntu-core";
+            debug("falling back to ubuntu-core instead of unavailable core snap");
+        }
+    }
+}
```

Comment:
I wonder if it makes sense to die() here if we fell all the way back to ubuntu-core but couldn't find it. I know we are checking that elsewhere so not a blocker. If you choose to leave as is can you add a comment that the existence check is done elsewhere?

Reply:
this doesn't seem to have been adressed